test/interactions/dragInteractionTests.ts

```diff
@@ -0,0 +1,140 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactions", () => {
+  describe("Drag", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    var startPoint = {
+      x: SVG_WIDTH / 4,
+      y: SVG_HEIGHT / 4
+    };
+    var endPoint = {
+      x: SVG_WIDTH / 2,
+      y: SVG_HEIGHT / 2
+    };
+
+    var outsidePointPos = {
+      x: SVG_WIDTH * 1.5,
+      y: SVG_HEIGHT * 1.5
+    };
+    var constrainedPos = {
+      x: SVG_WIDTH,
+      y: SVG_HEIGHT
+    };
+    var outsidePointNeg = {
+      x: -SVG_WIDTH / 2,
+      y: -SVG_HEIGHT / 2
+    };
+    var constrainedNeg = {
+      x: 0,
+      y: 0
+    };
+
+    it("onDragStart", () => {
+      var svg = generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var c = new Plottable.Component.AbstractComponent();
+      c.renderTo(svg);
+
+      var drag = new Plottable.Interaction.Drag();
+      var startCallbackCalled = false;
+      var receivedStart: Plottable.Point;
+      var startCallback = (p: Plottable.Point) => {
+        startCallbackCalled = true;
+        receivedStart = p;
+      };
+      drag.onDragStart(startCallback);
+      c.registerInteraction(drag);
+
+      var hitbox = c.hitBox();
+      triggerFakeMouseEvent("mousedown", hitbox, startPoint.x, startPoint.y);
+      assert.isTrue(startCallbackCalled, "callback was called on beginning drag (mousedown)");
+      assert.deepEqual(receivedStart, startPoint, "was passed the correct point");
+
+      startCallbackCalled = false;
+      triggerFakeMouseEvent("mousedown", hitbox, outsidePointPos.x, outsidePointPos.y);
+      assert.isFalse(startCallbackCalled, "does not trigger callback if drag starts outside the Component (positive) (mousedown)");
+      triggerFakeMouseEvent("mousedown", hitbox, outsidePointNeg.x, outsidePointNeg.y);
```

Comment:
Not sure if you need these outsidePointPos/Neg variables.  You could add a width / subtract a width or so instead and this would lose a variable